plugins/woocommerce/src/StoreApi/Routes/V1/CartUpdateCustomer.php

```diff
@@ -240,21 +240,8 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 	 * @return array
 	 */
 	protected function get_customer_billing_address( \WC_Customer $customer ) {
-		$validation_util = new ValidationUtils();
-		$billing_country = $customer->get_billing_country();
-		$billing_state   = $customer->get_billing_state();
-
 		$additional_fields = $this->additional_fields_controller->get_all_fields_from_object( $customer, 'billing' );
 
-		/**
-		 * There's a bug in WooCommerce core in which not having a state ("") would result in us validating against the store's state.
-		 * This resets the state to an empty string if it doesn't match the country.
-		 *
-		 * @todo Removing this handling once we fix the issue with the state value always being the store one.
-		 */
-		if ( ! $validation_util->validate_state( $billing_state, $billing_country ) ) {
-			$billing_state = '';
-		}
```

Comment:
This is no longer needed now that we reset at sanization level, I also tested this PR steps and they're still working

https://github.com/woocommerce/woocommerce-blocks/pull/8460